pig/src/main/java/org/jboss/pnc/bacon/pig/impl/pnc/PncEntitiesImporter.java

```diff
@@ -344,7 +344,14 @@ private SCMRepository getOrGenerateRepository(BuildConfig buildConfig) {
         }
 
         try {
-            return toStream(repoClient.getAll(matchUrl, null)).filter(buildConfig::matchesRepository).findAny();
+            List<SCMRepository> foundRepository = toStream(repoClient.getAll(matchUrl, null)).collect(Collectors.toList());
+            if (foundRepository.isEmpty()) {
+                return Optional.empty();
+            } else if (foundRepository.size() == 1) {
+                return Optional.of(foundRepository.get(0));
+            } else {
+                throw new RuntimeException("There exists more then one SCM Repository for url: " + matchUrl + " " + foundRepository);
```

Comment:
This should never happen (but I didn't want to write this message there to not embarass myself :D)